### ISSUE TITLE
spring-cloud-dataflow-deployer-local tests fail

### ISSUE THREAD

**Issue by maspen** [spring-cloud/spring-cloud-dataflow#421]
Am following http://cloud.spring.io/spring-cloud-dataflow/#quick-start steps.
When I run 'mvn clean -U package -e -P' on root-level project, get failed tests in spring-cloud-dataflow-deployer-local.

Attaching console output during 'package'-ing

Am on windows 7 & have redis (2.8.2400 64bit) server running.
[package.txt](https://github.com/spring-cloud/spring-cloud-dataflow/files/147037/package.txt)

**Comment by artembilan** [comment]
Well, for me that works well:
```
Results :

Tests run: 10, Failures: 0, Errors: 0, Skipped: 2

[INFO] 
[INFO] --- animal-sniffer-maven-plugin:1.14:check (enforce-java-7) @ spring-cloud-dataflow-deployer-local ---
[INFO] Checking unresolved references to org.codehaus.mojo.signature:java17:1.0
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 1:19.184s
[INFO] Finished at: Thu Feb 25 13:40:36 EST 2016
[INFO] Final Memory: 31M/557M
```
Although I tested only `spring-cloud-dataflow-deployer-local`.

I can't run your command from console because we don't know what is your `-P` :smile: 

Running the whole project with different `mnv` command...

Windows 7, too.

**Comment by artembilan** [comment]
Well, this is how does it looks in the end:
```
[INFO] --- maven-jar-plugin:2.5:jar (default-jar) @ spring-cloud-dataflow-docs ---
[WARNING] JAR will be empty - no content was marked for inclusion!
[INFO] Building jar: D:\SpringIO\spring-cloud\spring-cloud-dataflow\spring-cloud-dataflow-docs\target\spring-cloud-dataflow-docs-1.0.0.BUILD-SNAPSHOT.jar
[INFO] 
[INFO] --- maven-javadoc-plugin:2.10.1:jar (javadoc) @ spring-cloud-dataflow-docs ---
[INFO] 
[INFO] >>> maven-source-plugin:2.4:jar (attach-sources) @ spring-cloud-dataflow-docs >>>
[INFO] 
[INFO] <<< maven-source-plugin:2.4:jar (attach-sources) @ spring-cloud-dataflow-docs <<<
[INFO] 
[INFO] --- maven-source-plugin:2.4:jar (attach-sources) @ spring-cloud-dataflow-docs ---
[INFO] No sources in project. Archive not created.
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building spring-cloud-dataflow-dependencies 1.0.0.BUILD-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] spring-cloud-dataflow-parent ...................... SUCCESS [2.021s]
[INFO] spring-cloud-dataflow-core ........................ SUCCESS [3.966s]
[INFO] spring-cloud-dataflow-artifact-registry ........... SUCCESS [1.843s]
[INFO] spring-cloud-dataflow-deployer-spi ................ SUCCESS [2.657s]
[INFO] spring-cloud-dataflow-deployer-local .............. SUCCESS [1:02.560s]
[INFO] spring-cloud-dataflow-rest-resource ............... SUCCESS [1.847s]
[INFO] spring-cloud-dataflow-completion .................. SUCCESS [3.584s]
[INFO] spring-cloud-dataflow-server-core ................. SUCCESS [47.604s]
[INFO] spring-cloud-starter-dataflow-server-local ........ SUCCESS [1.109s]
[INFO] spring-cloud-dataflow-server-local ................ SUCCESS [23.186s]
[INFO] spring-cloud-dataflow-rest-client ................. SUCCESS [1.695s]
[INFO] spring-cloud-dataflow-shell ....................... SUCCESS [20.275s]
[INFO] spring-cloud-dataflow-docs ........................ SUCCESS [0.159s]
[INFO] spring-cloud-dataflow-dependencies ................ SUCCESS [0.000s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 2:52.844s
[INFO] Finished at: Thu Feb 25 13:47:21 EST 2016
[INFO] Final Memory: 90M/912M
[INFO] ------------------------------------------------------------------------
```
Maybe you should try the latest `master`?..

**Comment by maspen** [comment]
just cloned the master ()
"ran mvn clean" package at root directory & got this:

`[INFO] Scanning for projects...
[ERROR] The build could not read 1 project -> [Help 1]
[ERROR]   
[ERROR]   The project org.springframework.cloud:spring-cloud-dataflow-parent:1.0.0.BUILD-SNAPSHOT (C:\_software\spring-cloud-dataflow\spring-cloud-dataflow\pom.xml) has 1 error
[ERROR]     Non-resolvable import POM: Failure to find org.springframework.cloud:spring-cloud-task-dependencies:pom:1.0.0.M1 in http://repo.intellig.local:8081/artifactory/libs-release was cached in the local repository, resolution will not be reattempted until the update interval of central has elapsed or updates are forced @ line 92, column 16 -> [Help 2]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/ProjectBuildingException
[ERROR] [Help 2] http://cwiki.apache.org/confluence/display/MAVEN/UnresolvableModelException`

I have this problem when trying to access the spring dependencies.

then i run 'mvn clean package -e -P spring' (this is the profile name in the parent pom) & OutOfProcessModuleDeployerTests (like before) fails.

[package3.txt](https://github.com/spring-cloud/spring-cloud-dataflow/files/147146/package3.txt)

**Comment by artembilan** [comment]
Well, look into your error:

> http://repo.intellig.local:8081/artifactory/libs-release

That is definitely something wrong with your local Maven config. Maybe you have there some Nexus proxy...

When I run your `mvn clean package -e -P spring` I end up with:
```
[ERROR] Failed to execute goal org.codehaus.mojo:exec-maven-plugin:1.3.2:exec (imports) on project spring-cloud-dataflow-parent: Command execution failed. Cannot run program "egrep" (in directory "D:\SpringIO\spring-cloud\spring-cloud-dataflow"): CreateProcess e
rror=2, ?? ??????? ????? ????????? ???? -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.codehaus.mojo:exec-maven-plugin:1.3.2:exec (imports) on project spring-cloud-dataflow-parent: Command execution failed.
```

For me that works with the `mvn clean package -e -P !import`.
I'm sure that is fully different story.

You know... The `.m2` clean up helps sometimes :smile:

**Comment by maspen** [comment]
would you mind sharing your .m2/settings.xml?

**Comment by artembilan** [comment]
M-m-m. I have this in `<profiles>`:
````xml
<profile>
	<id>spring</id>
	<activation>
		<activeByDefault>true</activeByDefault>
	</activation>
	<repositories>
		<repository>
			<id>spring-snapshots</id>
			<name>Spring Snapshots</name>
			<url>http://repo.spring.io/libs-snapshot-local</url>
			<snapshots>
				<enabled>true</enabled>
			</snapshots>
		</repository>
		<repository>
			<id>spring-milestones</id>
			<name>Spring Milestones</name>
			<url>http://repo.spring.io/libs-milestone-local</url>
			<snapshots>
				<enabled>false</enabled>
			</snapshots>
		</repository>
		<repository>
			<id>spring-releases</id>
			<name>Spring Releases</name>
			<url>http://repo.spring.io/release</url>
			<snapshots>
				<enabled>false</enabled>
			</snapshots>
		</repository>
	</repositories>
	<pluginRepositories>
		<pluginRepository>
			<id>spring-snapshots</id>
			<name>Spring Snapshots</name>
			<url>http://repo.spring.io/libs-snapshot-local</url>
			<snapshots>
				<enabled>true</enabled>
			</snapshots>
		</pluginRepository>
		<pluginRepository>
			<id>spring-milestones</id>
			<name>Spring Milestones</name>
			<url>http://repo.spring.io/libs-milestone-local</url>
			<snapshots>
				<enabled>false</enabled>
			</snapshots>
		</pluginRepository>
	</pluginRepositories>
</profile>
````
Everything rest is by default - commented out.

**Comment by markpollack** [comment]
What version of maven are you using?  Using the mvnw.bat script should set up all the appropriate repo locations.

**Comment by styjdt** [comment]
Hi,
I also ran into this issue. After I finally got a setup of .m2 repository with proxy used for spring stuff beside another one working with corporate nexus, the tests also fail at my side. Following that to the logs of the module, which does not start, gives an issue with java.io.tmpdir which is somehow hardcoded under Windows in some of the Spring deploy magic.
See the output of the test:
[stderr_0.log.txt](https://github.com/spring-cloud/spring-cloud-dataflow/files/171577/stderr_0.log.txt)

I've created another issue directly describing the tmpdir issue: #421

Alexander
